tests/e2e/kcs/test_mon_crash_recovery_scenario.py

```diff
@@ -0,0 +1,126 @@
+    ROOK_CEPH_OPERATOR,
+    CEPHBLOCKPOOL,
+    STATUS_CLBO,
+)
+from ocs_ci.helpers.helpers import modify_deployment_replica_count
+from ocs_ci.ocs.resources.deployment import get_mon_deployments
+from ocs_ci.ocs.resources.pvc import get_pvc_objs
+from ocs_ci.ocs.resources.pod import get_ceph_tools_pod
+from ocs_ci.ocs.resources.storage_cluster import ceph_mon_dump
+from ocs_ci.framework.pytest_customization.marks import tier3
+from ocs_ci.ocs.defaults import OCS_OPERATOR_NAME
+from ocs_ci.helpers.helpers import wait_for_resource_state
+
+
+log = logging.getLogger(__name__)
+
+
+@tier3
+@pytest.mark.polarion_id("OCS-4942")
+@pytest.mark.bugzilla("2151591")
+class TestMonCrashRecoveryScenario:
+    @pytest.fixture(autouse=True)
+    def teardown_fixture(self, request):
+        def scale_up_deployments():
+            """Teardown function to scale deployments back to 1 replica."""
+            for dep in [OCS_OPERATOR_NAME, ROOK_CEPH_OPERATOR]:
+                log.info(f"Teardown: Scaling up {dep} to replica=1")
+                modify_deployment_replica_count(dep, 1)
+
+        request.addfinalizer(scale_up_deployments)
+
+    def test_mon_crash_recovery_scenario(self, pod_factory, request):
+        """
+        Verifies system behavior when a crash occurs in the mon-x deployment.
+
+        Steps:
+            1. Select a random mon and courrupt the mon database.
+            2. Start the IO workload in the background.
+            3. Scale down the deployments of ocs-operator,rook-ceph-operator and rook-ceph-mon-a.
+            4. Delete the Deployment of rook-ceph-mon-x and pvc rook-ceph-mon-x
+            5. Scale up the operators to replicas = 1
+            6. Verify 'ceph mon dump' command is working.
+            7. Check for the any crash has generated.
+        """
+
+        mon_obj = choice(get_mon_deployments())
+        mon_name = mon_obj.name
+        mon_pvc = mon_obj.data["metadata"]["labels"]["pvc_name"]
+        mon_pvc_obj = get_pvc_objs([mon_pvc])[0]
+        toolbox = get_ceph_tools_pod()
+        toolbox.exec_ceph_cmd("ceph crash archive-all")
+
+        # Step 1: Courrupting the mon database from the mon deployment.
+        mon_pods = mon_obj.pods
+        for monpod in mon_pods:
+            monpod.exec_cmd_on_pod(
+                f"rm -rf /var/lib/ceph/mon/ceph-{mon_name.split('-')[-1].strip()}",
+                ignore_error=True,
+            )
+            wait_for_resource_state(resource=monpod, state=STATUS_CLBO)
+
+        # Step 2:  Start IO Workload.
+        pod_obj = pod_factory(interface=CEPHBLOCKPOOL)
+
+        kwargs = {
+            "storage_type": "fs",
+            "size": "10G",
+            "runtime": 200,
+        }
+
+        io_thread = Thread(
+            target=pod_obj.run_io,
+            name="io_thread",
+            kwargs=kwargs,
+        )
+        io_thread.start()
+
+        # Step 3: Scale down the deployments of ocs-operator,rook-ceph-operator and rook-ceph-mon-a.
```

Comment:
we are picking some random mon name, please fix the hardcoded mon deployment name in the  comment message

Reply:
Done